fix: use set -e in create-pyxis-image task

mmalina opened the pull request:
Without this, the task execution would carry on even if something failed inside the script.

mmalina committed:
fix: use set -e in create-pyxis-image task

Without this, the task execution would carry on even if something
failed inside the script.

Signed-off-by: Martin Malina <mmalina@redhat.com>

jinqi7 commented:
/retest

johnbieren commented:
/test release-pipelines-e2e-suite

theflockers commented:
/test release-pipelines-e2e-suite

mmalina commented:
/retest

johnbieren commented:
/test release-pipelines-e2e-suite

jinqi7 commented:
/test release-pipelines-e2e-suite

mmalina commented:
/test release-pipelines-e2e-suite

mmalina commented:
/test release-pipelines-e2e-suite

johnbieren commented:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/konflux-ci_release-service-catalog/439/pull-ci-konflux-ci-release-service-catalog-development-release-pipelines-e2e-suite/1801252090434031616/artifacts/release-pipelines-e2e-suite/gather-extra/artifacts/pods/push-pyxis-managed-enej_managed-l9drf-create-pyxis-image-pod_step-create-pyxis-image.log

mmalina committed:
fix: run cleanup_tags in create-pyxis-image only if rhPush=true

The script expects to find a registry.access.redhat.com repository entry
for the image (which is only created if rhPush=true), so it would fail
for images that do not have one.

Signed-off-by: Martin Malina <mmalina@redhat.com>

mmalina commented:
> https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/konflux-ci_release-service-catalog/439/pull-ci-konflux-ci-release-service-catalog-development-release-pipelines-e2e-suite/1801252090434031616/artifacts/release-pipelines-e2e-suite/gather-extra/artifacts/pods/push-pyxis-managed-enej_managed-l9drf-create-pyxis-image-pod_step-create-pyxis-image.log

Thanks. I could have noticed myself :-\ So this is in fact something I missed when adding the cleanup script - the cleanup_tags script looks for a registry.access.redhat.com repository in the Container Image object and it will fail if it doesn't find one. But that repository is only created if rhPush=true. The rh-push-to-external-registry pipeline does contain the Pyxis stuff, but it runs the task with rhPush=false and that would cause the failure. I added a second commit with a fix. Let me know if you'd like to have a separate PR instead. (Also, the consequence is that tags are not cleaned up for this scenario. Right now I don't think it's a problem, because AFAIK these data are not really shown anywhere. But if we think we also need to clean up the tags here, I would need to modify the cleanup python script to work for this scenario.)

johnbieren commented:
> > https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/konflux-ci_release-service-catalog/439/pull-ci-konflux-ci-release-service-catalog-development-release-pipelines-e2e-suite/1801252090434031616/artifacts/release-pipelines-e2e-suite/gather-extra/artifacts/pods/push-pyxis-managed-enej_managed-l9drf-create-pyxis-image-pod_step-create-pyxis-image.log
> 
> Thanks. I could have noticed myself :-\ So this is in fact something I missed when adding the cleanup script - the cleanup_tags script looks for a registry.access.redhat.com repository in the Container Image object and it will fail if it doesn't find one. But that repository is only created if rhPush=true. The rh-push-to-external-registry pipeline does contain the Pyxis stuff, but it runs the task with rhPush=false and that would cause the failure. I added a second commit with a fix. Let me know if you'd like to have a separate PR instead. (Also, the consequence is that tags are not cleaned up for this scenario. Right now I don't think it's a problem, because AFAIK this data is not really shown anywhere. But if we think we also need to clean up the tags here, I would need to modify the cleanup python script to work for this scenario.)

Can you check with Scott to ensure it is okay to not cleanup the past tags with that pipeline? The code itself lgtm. And I am fine with it being one PR

mmalina commented:
> > > https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/konflux-ci_release-service-catalog/439/pull-ci-konflux-ci-release-service-catalog-development-release-pipelines-e2e-suite/1801252090434031616/artifacts/release-pipelines-e2e-suite/gather-extra/artifacts/pods/push-pyxis-managed-enej_managed-l9drf-create-pyxis-image-pod_step-create-pyxis-image.log
> > 
> > 
> > Thanks. I could have noticed myself :-\ So this is in fact something I missed when adding the cleanup script - the cleanup_tags script looks for a registry.access.redhat.com repository in the Container Image object and it will fail if it doesn't find one. But that repository is only created if rhPush=true. The rh-push-to-external-registry pipeline does contain the Pyxis stuff, but it runs the task with rhPush=false and that would cause the failure. I added a second commit with a fix. Let me know if you'd like to have a separate PR instead. (Also, the consequence is that tags are not cleaned up for this scenario. Right now I don't think it's a problem, because AFAIK this data is not really shown anywhere. But if we think we also need to clean up the tags here, I would need to modify the cleanup python script to work for this scenario.)
> 
> Can you check with Scott to ensure it is okay to not cleanup the past tags with that pipeline? The code itself lgtm. And I am fine with it being one PR

I talked to Scott and he agrees it makes sense to skip it there. FTR, the rh-push-to-external-registry pipeline (where we will skip the tag cleanup) is used by managed services.
